src/components/earthengine/Collection.js

```diff
@@ -51,31 +51,39 @@ export class CollectionSelect extends Component {
 
         if (id && !collections[id]) {
             loadEarthEngineCollection(id);
+        } else {
```

Comment:
For completeness I think this should be `else if (collections && collections[id])` to avoid a possible undefined object issue on line 70.

Can `collections` ever be null?  If not it should at least be required in `propTypes`.  If yes (or even if not) we should check it before de-referencing `collections[id]`

Reply:
I made collections required - the start value is set to an empty object in the store.